src/test.js

```diff
@@ -81,6 +88,34 @@ export function outputManyRuns(players, selectedDungeon, numRuns) {
   return output;
 }
 
+async function doRunsInWorker(team, waves, numRuns, progressCallback) {
+  let startedRuns = 0
+  let completedRuns = 0
+  let currentWorker = 0
+  let workerPromises = []
+
+  while (startedRuns < numRuns) {
+    startedRuns++;
+
+    if (currentWorker == 3) {
+      currentWorker = 0
+    } else {
+      currentWorker++
+    }
+
+    workerPromises.push(
+      workerPool[currentWorker].runDungeon(team, waves, 1)
```

Comment:
I'm running dungeons one at a time in each worker so we can get the progress callback to update the UI every time a run finishes. The workers otherwise support multiple runs if you wanted for some reason.

Note that usually you would use `postMessage` from the worker to report back progress, but I brought in the comlink library to abstract that away. Much easier to follow the worker code this way.